File: app/src/main/java/com/example/newyorktimearticles/domain/GetArticleUseCase.java
package com.example.newyorktimearticles.domain;

import com.example.newyorktimearticles.base.UseCase;
import com.example.newyorktimearticles.data.model.Result;
import com.example.newyorktimearticles.data.repository.GetArticlesRepository;
import com.example.newyorktimearticles.data.repository.IGetArticlesRepository;

import io.reactivex.Single;

public class GetArticleUseCase implements UseCase<Void, Single<Result>> {
    private IGetArticlesRepository getArticlesRepository;

    public GetArticleUseCase() {
        getArticlesRepository = new GetArticlesRepository();
    }

    @Override
    public Single<Result> execute(Void value) {
        return getArticlesRepository.getArticleList();
    }
}

File: app/src/main/java/com/gemidroid/newyorktimearticles/domain/GetArticleUseCase.kt
package com.gemidroid.newyorktimearticles.domain

import com.gemidroid.newyorktimearticles.data.model.Result
import com.gemidroid.newyorktimearticles.data.repository.IGetArticlesRepository
import kotlinx.coroutines.flow.Flow

class GetArticleUseCase(
    private val articleUseCase: IGetArticlesRepository
) {
    suspend fun getArticles(): Flow<Result> {
        return articleUseCase.getArticleList()
    }
}